addons/hr/models/hr_employee.py

```diff
@@ -362,7 +362,7 @@ def write(self, vals):
             self.message_unsubscribe(self.address_home_id.ids)
             if vals['address_home_id']:
                 self._message_subscribe([vals['address_home_id']])
-        if vals.get('user_id'):
+        if not vals.get('user_id', None) is None:
```

Comment:
Easier to read IMO :innocent: 

```suggestion
        if vals.get('user_id', None) is not None:
```
or if the goal is only to check that the key exists, regardless of its value:

```suggestion
        if 'user_id' in vals:
```

`in` is faster than `get()`
![image](https://github.com/odoo/odoo/assets/50135801/7fff1d5e-8ce6-4f87-9b28-4a137c318d93)